app/services/CheckoutService.scala

```diff
@@ -34,7 +35,19 @@ class CheckoutService(identityService: IdentityService, salesforceService: Sales
       userData <- userOrElseRegisterGuest
       memberId <- salesforceService.createOrUpdateUser(subscriptionData.personalData, userData.id)
       subscribeResult <- touchpointBackend.zuoraService.createSubscription(memberId, subscriptionData, ratePlan, paymentDelay)
-    } yield CheckoutResult(memberId, userData, subscribeResult)
+    } yield {
+      updateAuthenticatedUserDetails(subscriptionData.personalData, idUserOpt, authCookieOpt)
```

Comment:
Should we move this IO call between line 35 and 36? This would allow to execute the request in parallel with the Zuora and salesforce call, perhaps gaining a bit of performance